src/UI/Components/Diagram/actions.ts

```diff
@@ -431,91 +581,149 @@ export function appendEmbeddedEntity(
 }
 
 /**
- * Function that creates, appends and returns created Entity, differs from appendInstance by the fact that is used from the scope of Instance Composer and uses different set of data
+ * Populates a graph with default required entities derived from a service model.
  *
- * @param {dia.Graph} graph JointJS graph object
- * @param {dia.Paper} paper JointJS paper object
- * @param {ServiceModel} serviceModel that we want to base created entity on
- * @param {InstanceAttributeModel} entity created in the from
- * @param {boolean} isCore defines whether created entity is main one in given View
- * @param {string} holderName - name of the entity to which it is embedded/connected
- * @returns {ServiceEntityBlock} created JointJS shape
+ * @param {dia.Graph} graph - The jointJS graph to populate.
+ * @param {ServiceModel} serviceModel - The service model to use for populating the graph.
+ * @returns {void}
  */
-export function appendEntity(
+export function populateGraphWithDefault(
   graph: dia.Graph,
-  serviceModel: ServiceModel | EmbeddedEntity,
-  entity: InstanceAttributeModel,
-  isCore: boolean,
-  isEmbedded = false,
-  holderName = "",
-): ServiceEntityBlock {
-  //Create shape for Entity
-  const instanceAsTable = new ServiceEntityBlock().setName(serviceModel.name);
+  serviceModel: ServiceModel,
+): void {
+  //the most reliable way to get attributes default state is to use Field Creator
 
-  if (isEmbedded) {
-    instanceAsTable.setTabColor("embedded");
-  } else if (isCore) {
-    instanceAsTable.setTabColor("core");
-  }
+  const fieldCreator = new FieldCreator(new CreateModifierHandler());
+  const fields = fieldCreator.attributesToFields(serviceModel.attributes);
 
-  instanceAsTable.set("isEmbedded", isEmbedded);
-  instanceAsTable.set("holderName", holderName);
+  const attrs = createFormState(fields);
+  const isCore = true;
+  const isInEditMode = false;
 
-  if (
-    serviceModel.inter_service_relations &&
-    serviceModel.inter_service_relations.length > 0
-  ) {
-    instanceAsTable.set("relatedTo", new Map());
-  }
-  const attributesNames = serviceModel.attributes.map(
-    (attribute) => attribute.name,
+  const coreEntity = createComposerEntity(
+    serviceModel,
+    isCore,
+    isInEditMode,
+    attrs,
   );
 
-  appendColumns(instanceAsTable, attributesNames, entity);
-  //add to graph
-  instanceAsTable.addTo(graph);
+  coreEntity.addTo(graph);
 
-  moveCellFromColliding(graph, instanceAsTable);
+  const defaultEntities = addDefaultEntities(graph, serviceModel);
```

Comment:
> do something, then something else with the result
In my opinion it's not at all pattern of async without being async, I know we talked about it and I can't align with that thought, for me `.then` is useful when we doing real asynchronous operations, here everything is synched, we don't need to wait for anything. 

It's just a list of synchronous operations, that are tied to each other, and extracted for reusability and readability purposes.

Reply:
Sorry in advance for the long explanation, but I hope it will make you understand why I think it's important to not use the pattern you are applying here, and in other places. I don't perse expect to move to a promise, I'm just saying that the pattern you are using induces the understanding that this reads as one. 

I'll break the explanation down into these key points:

# 1. Separation of Concerns

The key issue here revolves around keeping responsibilities distinct. Clean architecture emphasizes that functions or components should do one thing and do it well. In the `populateGraphWithDefault` function, you're dealing with multiple responsibilities:

- Creating and adding entities to the graph.
- Handling connections between these entities.
- Managing layout (with DirectedGraph.layout).

Introducing `addDefaultEntities`, which is responsible for recursively adding entities, as a direct part of this flow and then performing further logic (like setting "embeddedTo" and connecting them) on those entities violates this principle. The logic related to embeddedTo and connectEntities could be encapsulated within addDefaultEntities itself or another dedicated function. This avoids creating intermediate results that need further manipulation outside the context of their primary function.

# 2. Potential Ambiguity Around Synchrony/Asynchrony

Even though the `addDefaultEntities` function is synchronous here, it could appear ambiguous to readers because it performs multiple steps that involve side effects:

- Dispatching events.
- Adding to the graph.
- Recursively adding more entities.

These actions might suggest to a future reader (or developer) that it could be a promise-based (asynchronous) function. When a function like this returns a result (`ServiceEntityBlock[]`), assigning it to a variable and then operating on it later (like modifying it or connecting entities) increases this ambiguity. It’s easy to misinterpret whether this action chain is synchronous or not, which introduces confusion.

**Better practice**: Hide these side effects within the function that generates or modifies the entities, and if more logic is required on the returned entities, do it within the function itself or return an entirely clean and completed result, not something that requires further steps.

# 3. Encapsulation and Single Responsibility in Functions

Clean architecture favors encapsulation. Ideally, when a function like `addDefaultEntities` is invoked, it should encapsulate all the actions related to adding default entities — including connecting them and setting any required properties (embeddedTo). By doing this, you avoid spreading logic across the calling function and keep related logic inside the function that’s responsible for handling it.

# 4. Avoiding Intermediate Steps with Side Effects

When you assign the result of `addDefaultEntities` to a variable (`defaultEntities`), you create a state that can lead to mutations (like setting properties) outside the function, in your case, it's after the function. This is not inherently bad on its own, but from a clean architecture perspective, **it's preferable that a function either fully mutates or fully returns without requiring follow-up steps that are detached from its purpose.**

Reply:
1. I understand that, I moved the logic for setting embeddedTo inside the `addDefaultEntities ` and renamed the function to `createdDefaultEmbeddedEntities` to better match its functionality.

2. I agree on part to hide side-effects as in 1.), and I'll try to incorporate it better across the functions, I still don't see how someone could interpret this code as asynchronous, but maybe I didn't came across the code that looks like this and is indeed async

3 and 4 I understand it as well, the actions that could/should be related to that function should be included. I will go through actions functions and I will try to search for this situations in other parts of code